app/controllers/api/v0/analytics_controller.rb

```diff
@@ -57,9 +57,12 @@ def load_owner
       end
 
       def validate_date_params
-        raise ArgumentError, "Required 'start' parameter is missing" if analytics_params[:start].blank?
+        if analytics_params[:start].blank?
+          raise ArgumentError,
+                I18n.t("api.v0.analytics_controller.start_missing")
+        end
 
-        message = "Date parameters 'start' or 'end' must be in the format of 'yyyy-mm-dd'"
+        message = I18n.t("api.v0.analytics_controller.date_format")
```

Comment:
Should we maybe call this `date_format_message`? The current name sounds a bit like this would be an actual data format. Or maybe we add one more level like `.errors.`?

Reply:
Sounds good. I'm still not sure where I should put error messages, they are sometimes in the `errors` directory, sometimes in controllers, or sometimes the same string across files. Your input is appreciated.

Reply:
`invalid_date_format`?

Reply:
Yeah, there are a few other inconsistencies, e.g. sometimes we use `something_icon` and sometimes `something.icon`. For now, maybe get things reasonably complete on the Ruby side, then have a look at all of these issues.